pkg/reconcilermanager/controllers/reconciler_base_test.go

```diff
@@ -513,13 +514,14 @@ func TestIsDeploymentSame(t *testing.T) {
 			isAutopilot:       false,
 			overrideLimit:     false,
 			manualChangeLimit: true,
-			// We do not define some container resource limits to make the resource burstable
-			// For these container resource limits, Config Sync only manage them when user
-			// define the resources limits through our resource override API.
-			// In this test case, the resource limits are not defined in original declared deployment,
-			// and user also not define the resource limits through our resource override API,
-			// so we will allow the manual limits change.
-			isSame: true,
+			// We do not define some containers' resource limits to make the resource burstable
+			// Config Sync only manage the resource limits when user
+			// define the resource limits through our resource override API.
+			// When user try to reset the override limits, the declared deployment will not have
+			// the limits defined. To make sure user can always successfully
+			// reset the resource limits override, we should keep tracking the resource limits differences
+			// between declared and current deployment, even there is no resource limits defined in declared deployment.
+			isSame: false,
```

Comment:
`isSame` doesn't make it clear this is the expected value we are asserting for in testcases. I'd suggest something like `expected`, `expectedSame`, etc

Reply:
Done, changed to expectedSame